ITK Build Failure when doing Python Wrapping using Make

<!-- The text within this markup is a comment, and is intended to provide
guidelines to open an issue for the ITK repository. This text will not
be part of the issue. -->


<!-- Before submitting an issue, please check that your issue has not been
already filed. -->

### Description

Build failure when building ITK with cmake using Make. It fails at around 41% progress with the following message.
```
[ 41%] Building CXX object Modules/Segmentation/Watersheds/src/CMakeFiles/ITKWatersheds.dir/itkWatershedMiniPipelineProgressCommand.cxx.o
make[2]: *** [Wrapping/Modules/ITKCommon/CMakeFiles/ITKCommonSwig.dir/build.make:461: Wrapping/Typedefs/itkFixedArray.i] Error 1
make[2]: *** Deleting file 'Wrapping/Typedefs/itkFixedArray.i'
make[1]: *** [CMakeFiles/Makefile2:32930: Wrapping/Modules/ITKCommon/CMakeFiles/ITKCommonSwig.dir/all] Error 2
```
I have tested both single thread and multi thread build with make (eg. make -j8). It fails for both cases.

It is building fine when using Ninja. 

Also I have narrowed down the change to commit: [ENH: Generate stub-files to support type hints](https://github.com/InsightSoftwareConsortium/ITK/commit/faa3ab0a7e36ad82ad9c817e282bea48e2ac380c)

<!-- Description of the bug. -->

### Steps to Reproduce
1. Configure cmake to use Make. 
2. Select Python wrapping
3. Build

<!--


Provide a minimal, complete, compilable, and verifiable example (commonly
abbreviated as MWE, Minimal Working Example, or sometimes referred to as SSEE,
Short, Self Contained, Correct (Compilable) Example, SSCCE) or code snippet,
either through a GitHub gist (https://gist.github.com/) or providing your own
files (including your source code, `CMakeLists.txt` file if applicable, and your
data) reproducing the issue or showing an incorrect result. -->

### Expected behavior

Build should succeed.

### Actual behavior

<!-- What actually happens. Include the relevant build error trace. -->
```none
```

### Reproducibility

<!-- What percentage of the time does it reproduce? -->

### Versions


<!-- If a tagged version, you can get this information by inspecting the
`ITK_VERSION_MAJOR` `ITK_VERSION_MINOR` and `ITK_VERSION_PATCH` variable
values in the `ITKConfig.cmake` file.

If the commit number is required, run `$ git rev-parse --short HEAD`. -->

### Environment

Tested on both Linux system and Mac system. It failed on both.

### Additional Information

<!-- Any additional information, configuration or data that might be necessary
to reproduce the issue. -->


<!-- **Note**: Use issues for their purpose; issues are not for code help. Need
help? Ask your question at ITK discourse (https://discourse.itk.org/). -->

--- a/CMake/ITKModuleExternal.cmake
+++ b/CMake/ITKModuleExternal.cmake
@@ -64,6 +64,9 @@ if(ITK_WRAPPING)
     # create the directory to avoid loosing case on windows
     file(MAKE_DIRECTORY ${ITK_WRAP_PYTHON_ROOT_BINARY_DIR})
 
+    set(ITK_STUB_DIR "${ITK_WRAP_PYTHON_ROOT_BINARY_DIR}/itk-stubs")
+    file(MAKE_DIRECTORY ${ITK_STUB_DIR})
+
     set(ITK_PYTHON_PACKAGE_DIR "${ITK_WRAP_PYTHON_ROOT_BINARY_DIR}/itk")
     # create the directory to avoid loosing case on windows
     file(MAKE_DIRECTORY ${ITK_PYTHON_PACKAGE_DIR})

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -260,6 +260,9 @@ if(ITK_WRAPPING)
     # create the directory to avoid loosing case on windows
     file(MAKE_DIRECTORY ${ITK_WRAP_PYTHON_ROOT_BINARY_DIR})
 
+    set(ITK_STUB_DIR "${ITK_WRAP_PYTHON_ROOT_BINARY_DIR}/itk-stubs")
+    file(MAKE_DIRECTORY ${ITK_STUB_DIR})
+
     set(ITK_PYTHON_PACKAGE_DIR "${ITK_WRAP_PYTHON_ROOT_BINARY_DIR}/itk")
     # create the directory to avoid loosing case on windows
     file(MAKE_DIRECTORY ${ITK_PYTHON_PACKAGE_DIR})

--- a/Wrapping/Generators/SwigInterface/CMakeLists.txt
+++ b/Wrapping/Generators/SwigInterface/CMakeLists.txt
@@ -317,7 +317,7 @@ macro(itk_end_wrap_module_swig_interface)
   set(module_interface_ext_file "${WRAPPER_MASTER_INDEX_OUTPUT_DIR}/${WRAPPER_LIBRARY_NAME}_ext.i")
   set(ITK_STUB_DIR "${ITK_DIR}/Wrapping/Generators/Python/itk-stubs")
 
-  set(ITK_STUB_PYI_FILE "${ITK_STUB_DIR}/${WRAPPER_LIBRARY_NAME}.pyi")
+  set(ITK_STUB_PYI_FILES )
 
   foreach(module ${SWIG_INTERFACE_MODULES})
     # create the swig interface
@@ -326,6 +326,8 @@ macro(itk_end_wrap_module_swig_interface)
     list(APPEND idx_files "${WRAPPER_MASTER_INDEX_OUTPUT_DIR}/${module}.idx")
     list(APPEND typedef_in_files "${WRAPPER_LIBRARY_OUTPUT_DIR}/${module}SwigInterface.h.in")
     list(APPEND typedef_files "${WRAPPER_MASTER_INDEX_OUTPUT_DIR}/${module}SwigInterface.h")
+    list(APPEND ITK_STUB_PYI_FILES "${ITK_STUB_DIR}/${module}.pyi")
+
 
     if(${module_prefix}_WRAP_EXPLICIT)
       list(APPEND opts --include "${WRAPPER_LIBRARY_NAME}Explicit.h")
@@ -365,8 +367,8 @@ macro(itk_end_wrap_module_swig_interface)
         unset(snake_case_config_file)
       endif()
 
-    add_custom_command(
-      OUTPUT ${i_files} ${typedef_files} ${idx_files} ${snake_case_config_file} ${ITK_STUB_PYI_FILE}
+      add_custom_command(
+      OUTPUT ${i_files} ${typedef_files} ${idx_files} ${snake_case_config_file} ${ITK_STUB_PYI_FILES}
       COMMAND ${Python3_EXECUTABLE} ${IGENERATOR}
         ${mdx_opts}
         ${swig_libs}

--- a/Wrapping/Generators/SwigInterface/igenerator.py
+++ b/Wrapping/Generators/SwigInterface/igenerator.py
@@ -10,6 +10,7 @@ from pathlib import Path
 from keyword import iskeyword
 from typing import List
 
+
 class ITKClass:
     def __init__(self, class_name):
         self.python_method_headers = {}
@@ -22,6 +23,7 @@ class ITKClass:
         self.is_enum = False
         self.has_superclass = False
         self.enums = []
+        self.submodule_name = ""
 
 
 def remove_class_type(itkclass: str) -> (str, bool):
@@ -1628,6 +1630,7 @@ if _version_info < (3, 7, 0):
             elif self.current_class not in self.classes:
                 self.classes[self.current_class] = ITKClass(self.current_class)
                 self.classes[self.current_class].typed = typed
+                self.classes[self.current_class].submodule_name = self.submoduleName
                 if typedef.name.endswith("Enums"):
                     self.classes[self.current_class].is_enum = True
                 if typedef.name.endswith("_Superclass"):
@@ -1725,22 +1728,26 @@ if _version_info < (3, 7, 0):
                 f.write(content)
 
 
-def write_submodule_pyi_file(
-    pyiFile: Path, submodule_name: str, header_code: str, interfaces_code: str
-) -> None:
-    # Write interface files to the stub directory to support editor autocompletions
+def init_submodule_pyi_file(pyiFile: Path, submodule_name: str) -> None:
     with open(pyiFile, "w") as pyiFile:
         pyiFile.write(
-            """#
+            f"""# Interface and Interface methods for submodule: {submodule_name}
 from typing import Union, Any
 # additional imports
 from .support.template_class import itkTemplate as _itkTemplate
 
 \n"""
         )
-        pyiFile.write(f"# Interface for submodule: {submodule_name}\n")
+
+
+def write_class_pyi(
+    pyiFile: Path, class_name: str, header_code: str, interfaces_code: str
+) -> None:
+    # Write interface files to the stub directory to support editor autocompletions
+    with open(pyiFile, "a+") as pyiFile:
+        pyiFile.write(f"# Interface for class: {class_name}\n")
         pyiFile.write(header_code)
-        pyiFile.write(f"# Interface methods for submodule: {submodule_name}\n")
+        pyiFile.write(f"# Interface methods for class: {class_name}\n")
         pyiFile.write(interfaces_code)
 
 
@@ -1983,9 +1990,6 @@ if __name__ == "__main__":
             swig_input_generator.snakeCaseProcessObjectFunctions
         )
 
-    outputPYIHeaderFile = StringIO()
-    outputPYIMethodFile = StringIO()
-
     classes = {}
 
     ordered_submodule_list: List[str] = []
@@ -2000,15 +2004,20 @@ if __name__ == "__main__":
 
     for submoduleName in ordered_submodule_list:
         generate_swig_input(submoduleName, classes)
+        init_submodule_pyi_file(
+            Path(f"{options.pyi_dir}/{submoduleName}.pyi"), submoduleName
+        )
 
-        for itk_class in classes.keys():
-            generate_class_pyi_def(
-                outputPYIHeaderFile, outputPYIMethodFile, classes[itk_class]
-            )
+    for itk_class in classes.keys():
+        outputPYIHeaderFile = StringIO()
+        outputPYIMethodFile = StringIO()
+        generate_class_pyi_def(
+            outputPYIHeaderFile, outputPYIMethodFile, classes[itk_class]
+        )
 
-        write_submodule_pyi_file(
-            Path(f"{options.pyi_dir}/{submoduleName}.pyi"),
-            submoduleName,
+        write_class_pyi(
+            Path(f"{options.pyi_dir}/{classes[itk_class].submodule_name}.pyi"),
+            itk_class,
             outputPYIHeaderFile.getvalue(),
             outputPYIMethodFile.getvalue(),
         )

